doc/cloud-init-daily-builds.md

```diff
@@ -0,0 +1,79 @@
+
+More information about build recipes can be found [on Launchpad](https://help.launchpad.net/Packaging/SourceBuilds/GettingStarted). Unfortunately, the documentation is sparse and somewhat outdated. Feel free to ask a cloud-init team member if you need help understanding or reproducing a build recipe.
+
+## When a build fails
+
+If you are subscribed to [build failure notifications](#build-failure-notifications), you should get an email when a daily build fails. When this happens, the daily PPA won't get updated, and Jenkins won't have the latest package to test.
+
+Every build comes with a build log. From the [recipe](https://code.launchpad.net/~cloud-init-dev/+recipe/cloud-init-daily-devel), the "Latest builds" section will have recent build status. The most recent should say "Failed to build". If you click that "Failed to build" link, you will be taken to a build status page. From there you can click the "buildlog" link. Failures are almost always 1 of 3 reasons:
+
+1. **Merge error**: Git is unable to merge the packaging branch with the main branch. If this happens, you'll see something like "Automatic merge failed; fix conflicts and then commit the result" near the bottom of the build log.
+2. **Quilt patch error**: If this happens, you'll see something like "Exception: Failed to apply quilt patches" at the bottom of the log.
+3. **Launchpad flakiness**: If this happens, you'll need to wait an re-run the build later.
+
+Fixing the build requires manually performing an upstream snapshot on the branch, and then fixing the problem as appropriate. See the [ubuntu release process docs](ubuntu_release_process.md#upstream-snapshot-process) for using the `new_upstream_snapshot.py` tool to do a manual snapshot.
+
+### Fixing a merge conflict
+
+As of this writing, there is a `new_upstream_snapshot.py` branch called `merge-strategy` that should handle the merge more intelligently and not require resolving any manual merge conflicts. If the branch is already merged, update this doc please XD.
+
+Alternatively, checkout the affected branch (we're assuming `devel` here), merge main, and resolve like you would any other merge conflict.
+```bash
+git checkout upstream/ubuntu/devel --track
+git merge main
+# See and fix merge conflict
+```
+
+### Fixing a quilt patch error
+
+If a quilt patch fails to apply, you need to manually refresh the quilt patch. This is easiest to do in conjunction with the `new_upstream_snapshot.py` script.
+
+```bash
+new_upstream_snapshot.py
+# Tool fails to apply a patch
+quilt push -f
+# Manually fix the patch, then run the next command
+quilt refresh
+# The following will ensure that the patches all apply and tests pass
+quilt pop -a
+tox
+quilt push -a
```

Comment:
push then tox then pop?

Reply:
Yeah, tests should pass with patches applied. Should I add a `-e py3` or is there another concern.

Reply:
I was trying to point out that the instructions look wrong - as written it is pop then tox then push.

Reply:
Oh! Yeah... I should fix that 🤦‍♂️